lib/zlib.js

```diff
@@ -78,6 +79,11 @@ function isInvalidStrategy(strategy) {
 }
 
 function zlibBuffer(engine, buffer, callback) {
+  // Streams do not support non-Buffer Uint8Arrays yet. Convert it to a
+  // Buffer without copying.
+  if (isUint8Array(buffer) && !Buffer.isBuffer(buffer))
```

Comment:
You can use the plain `Object.getPrototypeOf()` check here, that should give better performance in the common case and won’t hurt